datafusion/functions-nested/src/array_has.rs

```diff
@@ -94,7 +94,7 @@ impl Default for ArrayHas {
 impl ArrayHas {
     pub fn new() -> Self {
         Self {
-            signature: Signature::array_and_element(Volatility::Immutable),
+            signature: Signature::array_and_element(Volatility::Immutable, None),
```

Comment:
How about something like this, we don't necessary need to wrap into a util function, it is less helpful when there are many fields

```rust
            signature: Signature {
                type_signature: TypeSignature::ArraySignature(
                    ArrayFunctionSignature::Array {
                        arguments: vec![
                            ArrayFunctionArgument::Array,
                            ArrayFunctionArgument::Element,
                        ],
                        array_coercion: Some(ListCoercion::FixedSizedListToList),
                    },
                ),
                volatility: Volatility::Immutable,
            },
```

Reply:
but to avoid breaking `Signature::array_and_element` we can keep it. (without additional array_coercion field)

Reply:
I'm sorry, I don't think I fully understood what you're suggesting. I pushed https://github.com/apache/datafusion/pull/14532/commits/d404c5e61c209b03eed604271194348eaf2f0715 to try and address this, please let me know if that's what you had in mind.